app/src/main/kotlin/com/sergeylappo/booxrapiddraw/OverlayShowingService.kt

```diff
@@ -50,84 +54,69 @@ class OverlayShowingService : Service() {
+            stopSelf()
+            exitProcess(0)
+        }
+
+        val prefs = getSharedPreferences(PREFS_NAME, MODE_PRIVATE)
+        val isRunning = prefs.getBoolean(KEY_IS_RUNNING, false)
+
+        if (isRunning) {
+            Toast.makeText(this, "Terminating Rapid Draw Service...", Toast.LENGTH_SHORT).show()
+            stopSelf()
+            return START_NOT_STICKY // Prevents service from being recreated
+        }
+
+        // Set the flag to indicate that the service is now running
+        prefs.edit().putBoolean(KEY_IS_RUNNING, true).apply()
+
+        Toast.makeText(this, "Starting Rapid Draw Service", Toast.LENGTH_SHORT).show()
+        return START_STICKY // Service will be recreated if killed
+    }
+
+    private fun createForegroundNotification() {
+        val notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+
+        notificationManager.createNotificationChannel(
+            NotificationChannel(
+                CHANNEL_ID,
+                "Boox Rapid draw overlay service",
+                NotificationManager.IMPORTANCE_DEFAULT
+            )
         )
-        buttonParams.gravity = Gravity.START or Gravity.TOP
-        buttonParams.x = 0
-        buttonParams.y = 0
-        overlayButton = Button(this)
-        val gestureDetector = GestureDetector(this, ClickToStopServiceGestureDetector(this))
-        overlayButton.text = getString(R.string.exit)
-        overlayButton.setTextColor(Color.WHITE)
-
-        overlayButton.setOnTouchListener(object : OnTouchListener {
-            override fun onTouch(v: View, event: MotionEvent): Boolean {
-                if (gestureDetector.onTouchEvent(event)) {
-                    return true
-                }
-
-                if (event.action == MotionEvent.ACTION_MOVE || event.action == MotionEvent.ACTION_DOWN) {
-                    /* You can play around with the offset to set where you want the users finger to be on the view. Currently it should be centered.*/
-                    val xOffset = v.width / 2
-                    val yOffset = v.height / 2
-                    buttonParams.x = event.rawX.toInt() - xOffset
-                    buttonParams.y = event.rawY.toInt() - yOffset
-
-                    wm.updateViewLayout(overlayButton, buttonParams)
-                    return true
-                }
-                if (event.action == MotionEvent.ACTION_BUTTON_PRESS) {
-                    return false
-                }
-                return false
-            }
-        })
-        overlayButton.setBackgroundColor(Color.BLACK)
-        overlayButton.background.alpha = 100
 
-        wm.addView(overlayButton, buttonParams)
+        // add notification intent to finish the service
+        val pendingIntent = PendingIntent.getService(
+            this,
+            0,
+            Intent(this, OverlayShowingService::class.java).apply { action = "STOP" },
+            PendingIntent.FLAG_UPDATE_CURRENT
```

Comment:
I get the app crashing when testing this:
```
java.lang.RuntimeException: Unable to create service com.sergeylappo.booxrapiddraw.OverlayShowingService: java.lang.IllegalArgumentException: com.sergeylappo.booxrapiddraw: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
                                                                                                    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
```

```suggestion
            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
```